### DISCUSSION TITLE
-d tag有点混淆，求解

### DISCUSSION THREAD

**Discussion by sao321** [zfl9/chinadns-ng#205] [Q&A] [answered]
关于 -d, --default-tag <tag>              chn or gfw or <user-tag> or none(default)

如果我想国内的走国内DNS解析，国外的走国外DNS解析，这样使用是否正确？
chinadns-ng -l 5353 -c 223.5.5.5 -t 1.1.1.1 -g /tmp/gfwlist.txt -m /tmp/cdn.txt -M -d gfw -N
请问这里-M有必要使用吗？

如果-d chn，是不是就意味着gfw走国外，其它全部走国内？这时还有必须要使用-M吗？

请问我的理解有没有错误？实在抱歉读了readme依然没理清楚，谢谢！

**Comment by zfl9** [comment]
> readme应该说的比较清楚了。在域名列表一节有详细说明。

-d 选项只作用于 **未匹配任何域名列表** (也即*.txt文件)的域名。

-g选项给tag:gfw提供了域名列表
-m选项给tag:chn提供了域名列表
--group-dnl给自定义group提供了域名列表

-d xxx只对那些未匹配上述域名列表的域名有效。具体匹配逻辑在readme域名列表一节有，请阅读一遍。

---

关于你的疑问：

-d gfw存在时，-g gfw.txt没有意义。
-d chn存在时，-m chn.txt没有意义。
自定义group也是一样的道理。

**Reply by zfl9** [reply, marked as answer]
chinadns-ng -l 5353 -c 223.5.5.5 -t 1.1.1.1 -g /tmp/gfwlist.txt -m /tmp/cdn.txt -M -d gfw -N

此参数组合，等价于 **chnlist分流** (见readme的相关配置示例)，等价于以下参数组合：

chinadns-ng -l 5353 -c 223.5.5.5 -t 1.1.1.1 -m /tmp/cdn.txt -d gfw -N

**Comment by zfl9** [comment]
也许早期readme文档不太完善，建议重新看一下新的readme，尤其是 配置示例，域名列表 这几节。

**Reply by sao321** [reply]
感谢大佬恢复，我不是程序员，readme读了N遍越读越晕。我想要做的非常简单，就是国内走国内DNS，国外走国外DNS解析。

大佬回复说”-d xxx只对那些未匹配上述域名列表的域名有效“，可是如果我去掉-d gfw，即便匹配了-g -m两项，使用[https://browserleaks.com/dns](https://browserleaks.com/dns)测试，可以看到国内的DNS解析全部泄露，但是配上了-d gfw后，则只会显示国外DNS。因为看到了readme里说
”-m chnlist.txt -d gfw：chn列表的域名走国内上游，其他走可信上游。“所以加上了-d gfw部分。
用大佬的
chinadns-ng -l 5353 -c 223.5.5.5 -t 1.1.1.1 -m /tmp/cdn.txt -d gfw -N
又试了一下，确实效果似乎一样，非常感谢。

至于配置示例那部分，我是越看越迷糊完全不理解，实在抱歉打扰大佬回复像我这样的小白。
